paasta_tools/api/api.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+# Copyright 2015-2016 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Responds to paasta service and instance requests.
+"""
+import argparse
+import logging
+import sys
+
+import requests_cache
+from gevent.wsgi import WSGIServer
+from pyramid.config import Configurator
+
+from paasta_tools.api import settings
+from paasta_tools.utils import load_system_paasta_config
+
+
+log = logging.getLogger(__name__)
+
+
+def parse_paasta_api_args():
+    parser = argparse.ArgumentParser(description='Runs a PaaSTA API server')
+    parser.add_argument(
+        '-D', '--debug',
+        dest='debug',
+        action='store_true', default=False,
+        help="output the debug logs"
+    )
+    parser.add_argument(
+        'port', type=int,
+        help="port number for the api server")
+    parser.add_argument(
+        '-d', '--soa-dir',
+        dest="soa_dir",
+        help="define a different soa config directory"
+    )
+    args = parser.parse_args()
+    return args
+
+
+def main(argv=None):
+    args = parse_paasta_api_args()
+    if args.debug:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.WARNING)
+
+    if args.soa_dir:
+        settings.soa_dir = args.soa_dir
+    settings.cluster = load_system_paasta_config().get_cluster()
+
+    # Set up transparent cache for http API calls. With expire_after, responses
+    # are removed only when the same request is made. Expired storage is not a
+    # concern here. Thus remove_expired_responses is not needed.
+    requests_cache.install_cache("paasta-api", backend="memory", expire_after=30)
+
+    config = Configurator()
+    config.add_route('service.instance.status', '/v1/{service}/{instance}/status')
+    config.add_route('service.list', '/v1/{service}')
+    config.scan()
+
+    app = config.make_wsgi_app()
```

Comment:
Can you make returning the app another method, this allows us to pass the app to something like `uwsgi`/`gunicorn` in case we don't want to run it with `WSGIServer` in the future. 

Example: https://github.com/Pylons/shootout/blob/master/shootout/__init__.py#L13-L39